rules-java-archives/addon/src/main/java/org/jboss/windup/rules/apps/java/archives/identify/SortedFileArchiveIdentifier.java

```diff
@@ -0,0 +1,277 @@
+
+        do {
+            if (range.diff() < 300) // The longest entry length estimation * 2.
+                return lookForEntry_LinearSearch(sha1Hash, seekable, range);
+
+            long pivot = (range.a + range.b) / 2;
+
+
+            seekable.seek(pivot);
+            //findNewLine(seekable);
+            String line = seekable.readLine();
+            if (line == null) // EOF
+                return null;
+
+            // We may or may not hit a start of line.
+            if (!isValidSHA1AndGAV(line))
+            {
+                line = seekable.readLine();
+                if (line == null) // EOF
+                    return null;
+                if (!isValidSHA1AndGAV(line))
+                    throw new IllegalArgumentException("Invalid SHA1 GAV entry found:\n\t" + line);
+            }
+
+            // Compare the two SHA1 hashes (by bytes).
+            System.arraycopy(line.substring(0, SHA1_LENGTH).getBytes(), 0, entryHashBytes, 0, SHA1_LENGTH);
+            int compare = compare(hashBytes, entryHashBytes);
+            if (compare == 0)
+                // Match
+                return line;
+            if (compare < 0) {
+                range.b = pivot;
+            }
+            else {
+                range.a = pivot;
+            }
+
+            if (log.isLoggable(Level.FINEST))
+                log.finest("Range: " + range);
+
+        } while (true);
+    }
+
+
+    //UnsignedBytes.lexicographicalComparator()  needs Guava
+    public static int compare(byte[] left, byte[] right) {
+        for (int i = 0, j = 0; i < left.length && j < right.length; i++, j++) {
+            int a = (left[i] & 0xff);
+            int b = (right[j] & 0xff);
+            if (a != b) {
+                return a - b;
+            }
+        }
+        return left.length - right.length;
+    }
+
+
+
+    private static String lookForEntry_LinearSearch(String sha1hash, RandomAccessFile seekable, Range range) throws IOException
+    {
+        seekable.seek(range.a);
+        while (true) {
+            // Reached the end of range?
+            if (seekable.getFilePointer() >= range.b)
+                return null;
+
+            String line = seekable.readLine();
+            if (line == null) // EOF
+                return null;
+
+            // We may or may not hit the start of the line on the first turn.
+            if (!isValidSHA1AndGAV(line))
+                continue;
+
+
+            // Compare the two SHA1 hashes (by bytes).
+            /*System.arraycopy(line.substring(0,SHA1_LENGTH).getBytes(), 0, entryHashBytes, 0, SHA1_LENGTH);
+            int compare = compare(hashBytes, entryHashBytes);
+            if (compare == 0)
+                // Match
+                return line;
+            */
+            if (0 == sha1hash.compareToIgnoreCase(line.substring(0, SHA1_LENGTH)))
+                return line;
+        }
+
+    }
+
+
+    private static final String ID = "([\\w-]+(\\.[\\w-]+)*)";
+
+    // G:A[:P[:C]]:V
```

Comment:
I think that in context of Maven coorinates it's quite clear, especially if the addon is full of SHA1toGAV etc. But ok